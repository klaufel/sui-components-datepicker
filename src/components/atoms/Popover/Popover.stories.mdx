import {Meta, Props, Story, Preview} from '@storybook/addon-docs/blocks'
import Popover from '.'

<Meta title="Atoms/Popover" component={Popover} />

# Popover

Let's define a story for our `Popover` component:

<Preview>
  <Story name="default">
    <Popover isActive>
      <div>...children</div>
    </Popover>
  </Story>
</Preview>

## Props

<Props of={Popover} />
