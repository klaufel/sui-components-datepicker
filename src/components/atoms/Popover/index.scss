.Popover {
  position: absolute;
  top: 100%;
  background: #fff;
  z-index: 10;
  padding: 12px;
  box-shadow: 0 0 5px 0 rgba(0, 0, 0, 0.75);
  opacity: 0;
  visibility: hidden;
  transition: opacity .2s, visibility .2s;
  transition-delay: .25s;

  &.is-active {
    opacity: 1;
    visibility: visible;
    transition-delay: 0s;
  }
}
