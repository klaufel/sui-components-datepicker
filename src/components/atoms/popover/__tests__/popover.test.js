import React from 'react'
import {render} from '@testing-library/react'
import Popover from '..'

describe('Popover', () => {
  test('it works default', () => {
    const {container} = render(
      <Popover>
        <div>...children</div>
      </Popover>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('it works with is open', () => {
    const {container} = render(
      <Popover isOpen>
        <div>...children</div>
      </Popover>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('it works with with close on click outisde', () => {
    const {container} = render(
      <Popover isOpen closeOutside>
        <div>...children</div>
      </Popover>
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
