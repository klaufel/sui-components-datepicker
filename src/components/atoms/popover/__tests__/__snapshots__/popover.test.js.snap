// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover it works default 1`] = `
<div
  class="Popover"
>
  <div>
    ...children
  </div>
</div>
`;

exports[`Popover it works with is open 1`] = `
<div
  class="Popover is-active"
>
  <div>
    ...children
  </div>
</div>
`;

exports[`Popover it works with with close on click outisde 1`] = `
<div
  class="Popover is-active"
>
  <div>
    ...children
  </div>
</div>
`;
